src/wh_cryptocb.c

```diff
@@ -0,0 +1,474 @@
+                /* read out */
+                else {
+                    XMEMCPY(info->pk.eccsign.out, out, packet->pkEccSignRes.sz);
+                    *info->pk.eccsign.outlen = packet->pkEccSignRes.sz;
+                }
+            }
+            break;
+        case WC_PK_TYPE_ECDSA_VERIFY:
+            /* sig and hash are after the fixed size fields */
+            sig = (uint8_t*)(&packet->pkEccVerifyReq + 1);
+            hash = (uint8_t*)(&packet->pkEccVerifyReq + 1) +
+                info->pk.eccverify.siglen;
+            /* set len */
+            packet->len = sizeof(packet->pkEccVerifyReq) +
+                info->pk.eccverify.siglen + info->pk.eccverify.hashlen;
+            /* set keyId */
+            packet->pkEccVerifyReq.keyId = info->pk.eccverify.key->devCtx;
+            /* set curveId */
+            packet->pkEccVerifyReq.curveId =
+                wc_ecc_get_curve_id(info->pk.eccverify.key->idx);
+            /* set sig and hash sz */
+            packet->pkEccVerifyReq.sigSz = info->pk.eccverify.siglen;
+            packet->pkEccVerifyReq.hashSz = info->pk.eccverify.hashlen;
+            /* copy sig and hash */
+            XMEMCPY(sig, info->pk.eccverify.sig, info->pk.eccverify.siglen);
+            XMEMCPY(hash, info->pk.eccverify.hash, info->pk.eccverify.hashlen);
+            /* write request, read response on same memory */
+            ret = hsmClientXferPacket(ctx);
+            if (ret == 0) {
+                if (packet->type == WOLFHSM_ERROR)
+                    ret = packet->error;
+                /* read res */
+                else
+                    *info->pk.eccverify.res = packet->pkEccVerifyRes.res;
+            }
+            break;
+        case WC_PK_TYPE_EC_CHECK_PRIV_KEY:
+            /* set len */
+            packet->len = sizeof(packet->pkEccCheckReq);
+            /* set keyId */
+            packet->pkEccCheckReq.keyId = info->pk.ecc_check.key->devCtx;
+            /* set curveId */
+            packet->pkEccCheckReq.curveId =
+                wc_ecc_get_curve_id(info->pk.eccverify.key->idx);
+            /* write request, read response on same memory */
+            ret = hsmClientXferPacket(ctx);
+            if (ret == 0 && packet->type == WOLFHSM_ERROR)
+                ret = packet->error;
+            break;
+        case WC_PK_TYPE_NONE:
+        default:
+            ret = CRYPTOCB_UNAVAILABLE;
+            break;
+        }
+        break;
+#endif
+    case WC_ALGO_TYPE_RNG:
+        /* out is after the fixed size fields */
+        out = (uint8_t*)(&packet->rngRes + 1);
+        /* set sz */
+        packet->rngReq.sz = info->rng.sz;
+        /* write request */
+        ret = wh_Client_SendRequest(ctx, group, WC_ALGO_TYPE_RNG,
+            sizeof(packet->rngReq), rawPacket);
+        if (ret == 0) {
+            do {
+                ret = wh_Client_RecvResponse(ctx, &group, &action, &dataSz,
+                    rawPacket);
+                sleep(1);
```

Comment:
This is not portable.  Let's replace this with XUSLEEP(1000) macro that the user can replace.  Default implementation simply returns

Reply:
Added to todo list